packages/ember-metal/lib/meta.js

```diff
@@ -262,7 +262,7 @@ export class Meta {
   hasDeps(subkey) {
     let pointer = this;
     while (pointer !== undefined) {
-      if (pointer._deps && pointer._deps[subkey]) {
+      if (pointer._deps !== undefined && pointer._deps[subkey] !== undefined) {
```

Comment:
I'd suggest as follow-up cleanup to avoid the pointer._deps reload, i.e.

```javascript
let pointerDeps = pointer._deps;
if (pointerDepts !== undefined && pointerDeps[subkey] !== undefined) {
  return true;
}
```